components/clp-credential-manager/src/config.rs

```diff
@@ -33,42 +34,18 @@ impl AppConfig {
     /// * Returns [`ServiceError::Io`] if the file cannot be read.
     /// * Returns [`ServiceError::Yaml`] if parsing fails.
     pub fn from_file(path: &Path) -> ServiceResult<Self> {
-        let contents = fs::read_to_string(path)?;
-        let config: Self = serde_yaml::from_str(&contents)?;
-        Ok(config)
+        yaml::from_path(path).map_err(map_yaml_error)
```

Comment:
```suggestion
        yaml::from_path(path)?
```
Also, it might be easier to let main call ` yaml::from_path`
For the modularity, `config.rs` only defines the data format and shouldn't care about whether the data is deserialized from yaml or json.

Reply:
but feel free to keep the code as-is. Different people would have different opinions on this.